taichi/program/program.cpp

```diff
@@ -569,6 +569,17 @@ void Program::delete_ndarray(Ndarray *ndarray) {
   ndarrays_.erase(ndarray);
 }
 
+void Program::fill_ndarray_fast(Ndarray *ndarray, uint32_t val) {
```

Comment:
C++ nerd: I'm thinking about tag dispatching here LOL. One way to do it would be have a separate Ndarray class for each backend. (This should work, because by the time we allocate an Ndarray, we know which backend it runs on). But this is nothing urgent at all.

Reply:
agreed!